pkg/types/libvirt/validation/machinepool.go

```diff
@@ -8,5 +8,14 @@ import (
 
 // ValidateMachinePool checks that the specified machine pool is valid.
 func ValidateMachinePool(p *libvirt.MachinePool, fldPath *field.Path) field.ErrorList {
-	return field.ErrorList{}
+	allErrs := field.ErrorList{}
+
+	if p.DomainMemory < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("DomainMemory"), p.DomainMemory, "Domain Memory (MiB) must be positive"))
+	}
+	if p.DomainVcpu < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("DomainVcpu"), p.DomainVcpu, "Domain VCPU count must be positive"))
```

Comment:
technically `0` is also not *positive* :P

Reply:
yep, zero is fine: in that case, the builtin default should be used.